### ISSUE TITLE
G-API: test input data generation: undefined behavior

### ISSUE THREAD

**Issue by alalek** [opencv/opencv#12924]
"Scalar" values in tests are reversed due [UB in evaluation order of function parameters](https://stackoverflow.com/questions/2934904/order-of-evaluation-in-c-function-parameters) (x86 - R->L, ppc - L->R).
```.diff
sc = [11, 93, 48, 52]  // x86
sc = [52, 48, 93, 11]  // ppc (qemu)
```
    To fix that need to apply this patch (to force L->R order):
```.diff
--- a/modules/gapi/test/common/gapi_tests_common.hpp
+++ b/modules/gapi/test/common/gapi_tests_common.hpp
@@ -37,7 +37,11 @@ public:
         in_mat2 = cv::Mat(sz_in, type);
 
         auto& rng = cv::theRNG();
-        sc = cv::Scalar(rng(100),rng(100),rng(100),rng(100));
+        double s1 = rng(100);
+        double s2 = rng(100);
+        double s3 = rng(100);
+        double s4 = rng(100);
+        sc = cv::Scalar(s1, s2, s3, s4);
         cv::randu(in_mat1, cv::Scalar::all(0), cv::Scalar::all(255));
         cv::randu(in_mat2, cv::Scalar::all(0), cv::Scalar::all(255));
 
@@ -53,7 +57,11 @@ public:
         in_mat1 = cv::Mat(sz_in, type);
 
         auto& rng = cv::theRNG();
-        sc = cv::Scalar(rng(100),rng(100),rng(100),rng(100));
+        double s1 = rng(100);
+        double s2 = rng(100);
+        double s3 = rng(100);
+        double s4 = rng(100);
+        sc = cv::Scalar(s1, s2, s3, s4);
 
         cv::randu(in_mat1, cv::Scalar::all(0), cv::Scalar::all(255));
```

details #12830

**Comment by dmatveev** [comment]
Good catch, thanks!

**Comment by dmatveev** [comment]
Seems the issue has been particularly (or totally) fixed by #12944 

Let me double-check if all affected core has been really addressed.
